tests/components.test.js

```diff
@@ -91,54 +97,37 @@ it("Doesn't call `onChange` when user changes a hue of a grayscale color", () =>
   const { container } = render(<HexColorPicker color="#000" onChange={handleChange} />);
   const hue = container.querySelector(".react-colorful__hue .react-colorful__interactive");
 
-  fireEvent.touchStart(hue, { touches: [{ pageX: 0, pageY: 0 }] });
-  fireEvent.touchMove(hue, { touches: [{ pageX: 100, pageY: 0 }] });
+  fireEvent.pointerDown(hue, { pageX: 0, pageY: 0 });
+  fireEvent.pointerMove(hue, { pageX: 100, pageY: 0 });
 
   expect(handleChange).not.toHaveBeenCalled();
 });
 
-it("Triggers `onChange` after a mouse interaction", async () => {
+it("Triggers `onChange` after a pointer interaction", async () => {
   const handleChange = jest.fn();
   const result = render(<RgbaColorPicker onChange={handleChange} />);
   const saturation = result.container.querySelector(
     ".react-colorful__saturation .react-colorful__interactive"
   );
 
-  fireEvent(saturation, new FakeMouseEvent("mousedown", { pageX: 0, pageY: 0 }));
-  fireEvent(saturation, new FakeMouseEvent("mousemove", { pageX: 10, pageY: 10 }));
+  fireEvent.pointerDown(saturation, { pageX: 0, pageY: 0 });
+  fireEvent.pointerMove(saturation, { pageX: 10, pageY: 10 });
 
   expect(handleChange).toHaveReturned();
 });
 
-it("Triggers `onChange` after a touch interaction", async () => {
+it("Triggers `onChange` after a pointer interaction", async () => {
   const handleChange = jest.fn((hex) => hex);
   const initialValue = { h: 0, s: 100, v: 100 };
   const result = render(<HsvColorPicker color={initialValue} onChange={handleChange} />);
   const hue = result.container.querySelector(".react-colorful__hue .react-colorful__interactive");
 
-  fireEvent.touchStart(hue, { touches: [{ pageX: 0, pageY: 0, bubbles: true }] });
-  fireEvent.touchMove(hue, { touches: [{ pageX: 55, pageY: 0, bubbles: true }] });
+  fireEvent.pointerDown(hue, { pageX: 0, pageY: 0 });
+  fireEvent.pointerMove(hue, { pageX: 55, pageY: 0 });
 
   expect(handleChange).toHaveReturnedWith({ h: 180, s: 100, v: 100 });
 });
 
-it("Pointer doesn't follow the mouse if it was released outside of the document bounds", async () => {
-  const handleChange = jest.fn();
-  const result = render(<RgbaColorPicker onChange={handleChange} />);
-  const saturation = result.container.querySelector(
-    ".react-colorful__saturation .react-colorful__interactive"
-  );
-
-  // User presses and moves the cursor inside the window
-  fireEvent(saturation, new FakeMouseEvent("mousedown", { pageX: 20, pageY: 10 })); // 1
-  fireEvent(saturation, new FakeMouseEvent("mousemove", { pageX: 10, pageY: 10 })); // 2
-  // User releases the mouse button outside of the document bounds with no `mouseup` event fired
-  // User moves the cursor back to the document with no button pressed
-  fireEvent(saturation, new FakeMouseEvent("mousemove", { pageX: 1, pageY: 50, buttons: 0 })); // 3
-
-  expect(handleChange).toHaveReturnedTimes(2); // the last `mousemove` has to be ignored
-});
```

Comment:
fulfilled by using pointerevents

Reply:
Can we emulate the same test case (pointer outside of the window) with `PointerEvent`?